### PR TITLE
mail/notmuch: Update to version 0.29.2

### PR DESCRIPTION
**Pull request by rseichter**
Bugfix for file descriptor leak when opening gzipped mail files.

### CODE REVIEW

**Comment by mf2k** [comment]
The commit message should be updated to remove the category:

`notmuch: Update to version 0.29.2
`

File: mail/notmuch/Portfile [hunk 1]
```diff
@@ -51,9 +49,6 @@ post-patch {
 
 conflicts_build     ${name} xcbuild
 
-compiler.blacklist-append *gcc-3.* *gcc-4.* {clang < 300} llvm-g++-4.2
-compiler.whitelist clang macports-clang-5.0 
-
```
Comment (mf2k) [inline mail/notmuch/Portfile]: Why are these being removed?

File: mail/notmuch/Portfile [hunk 2]
```diff
@@ -5,8 +5,7 @@ PortGroup           compiler_blacklist_versions 1.0
 PortGroup           conflicts_build             1.0
 
 name                notmuch
-version             0.29.1
-revision            1
```
Comment (mf2k) [inline mail/notmuch/Portfile]: We now prefer to keep the `revision` line, but set it back to 0.

Comment (rseichter) [reply mail/notmuch/Portfile]: We do? https://guide.macports.org/#reference.keywords does not mention this, `revision` is documented as optional.

File: mail/notmuch/Portfile (hunk 1 above)
Comment (rseichter) [reply mail/notmuch/Portfile]: Because I want to see where in the hosted builds possible errors occur.

**Comment by rseichter** [comment]
> The commit message should be updated to remove the category

Ah, old Gentoo Linux habit.

File: mail/notmuch/Portfile (hunk 2 above)
Comment (mf2k) [reply mail/notmuch/Portfile]: There was a discussion on the mailing list about it not too long ago. The guide is behind on this.

Comment (rseichter) [reply mail/notmuch/Portfile]: I try to follow the ML closely, but I don't remember this being discussed. Would you perhaps be able to point me to the relevant thread? I actually use `notmuch` for my mail, and it is pretty good at searching, but I could not find the relevant emails.

**Comment by rseichter** [comment]
Revision and commit comment updated.

**Pull request merged**